vscode-lean4/src/leanclient.ts

```diff
@@ -100,17 +102,35 @@ export class LeanClient implements Disposable {
     }
 
     async showRestartMessage(): Promise<void> {
-        const restartItem = 'Restart Lean Language Server';
-        const item = await window.showErrorMessage('Lean Language Server has stopped unexpectedly.', restartItem)
-        if (item === restartItem) {
-            void this.start();
+        if (!this.showingRestartMessage) {
+            this.showingRestartMessage = true;
+            const restartItem = 'Restart Lean Server';
+            const item = await window.showErrorMessage('Lean Server has stopped unexpectedly.', restartItem)
+            this.showingRestartMessage = false;
+            if (item === restartItem) {
+                void this.start();
+            }
+        }
+    }
+
+    async showRestartFileMessage(): Promise<void> {
+        if (!this.showingRestartFileMessage) {
+            this.showingRestartFileMessage = true;
+            const restartItem = 'Restart the Lean Server on this file';
+            const item = await window.showErrorMessage('The Lean Server has stopped processing this file.', restartItem)
+            this.showingRestartFileMessage = false;
+            if (item === restartItem) {
+                if (window.activeTextEditor) {
+                    return this.restartFile(window.activeTextEditor.document);
```

Comment:
I think await should be here instead of return.

Reply:
Ok, makes sense